gapis/api/vulkan/vulkan.go

```diff
@@ -78,6 +78,137 @@ func (s *State) SetupInitialState(ctx context.Context, state *api.GlobalState) {
 	}
 }
 
+// TrimInitialState scans the capture commands to see which parts of the initial
+// state are actually used, and removes some unused parts from it.
+//
+// Note: the current approach consists in "manually" monitoring which Vulkan
+// objects are being used in callbacks passed to sync.MutateWithSubcommands,
+// however this basically re-encode some state tracking logic found in the API
+// files. A better way would be to use an api.StateWatcher and rely on api.RefID
+// to track which objects are accessed: this would avoid to re-encode state
+// tracking logic here. There might be some pitfalls though, e.g. when a command
+// just reads the handle of an object, the state watcher would not mark an
+// access to that object. For instance, when creating a derivate pipeline, a
+// VkPipeline handle is used in BasePipelineHandle, the API implementation reads
+// this handle, but does not access the corresponding object. So using
+// api.StateWatcher might need some wider design considerations.
+func (s *State) TrimInitialState(ctx context.Context, capturePath *path.Capture) error {
+	// Parts of the state we want to record the usage of.
+	descriptorSets := map[VkDescriptorSet]struct{}{}
```

Comment:
Should not this be `interface{}` instead `struct{}`?

Reply:
This is a regular way to have a set in Go. Another classical approach is a map of booleans. I think using a map of `struct{}` is the regular way to have sets in our codebase. You can try `git grep 'struct{}{}` to see some examples.

Using `interface{}` is probably not a good idea since you could put anything in the map. With `struct{}`, the only thing you can put is `struct{}{}`.